[TSVB] Removes less support from markdown editor

VladLasitsa opened the pull request:
Closes: #100488, Closes: #78422

## Summary

- Removed `markdown_less` from saved_object.
- Removed markdown id from `markdown_css`
- Added migration script for first two items.
- Used `@emotion/react` for encapsulationing styles inside markdown instead of using id.

VladLasitsa commented on src/plugins/visualizations/server/migrations/visualization_saved_object_migrations.ts:
```diff
@@ -1102,4 +1136,5 @@ export const visualizationSavedObjectTypeMigrations = {
     migrateTagCloud,
     replaceIndexPatternReference
   ),
+  '7.15.0': flow(removeMarkdownLessFromTSVB),
```
We cannot skip version. Maybe someone can provide idea how we should add migration script for 8.0.0?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/)
* Commit: 8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute oss-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/execution/node/584/log)
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/execution/node/592/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/","number":"150558","commit":"8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e"}]}
PIPELINE-->

VladLasitsa commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/)
* Commit: 20c9bf9b9b06aab4d0f95024db35582ec683f276
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/20c9bf9b9b06aab4d0f95024db35582ec683f276)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/execution/node/695/log)
* [Execute oss-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/execution/node/584/log)

### History
* :broken_heart: [Build #150558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/) failed 8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/","number":"150558","commit":"8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/","number":"150573","commit":"20c9bf9b9b06aab4d0f95024db35582ec683f276"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/)
* Commit: 1ccd7d1b4ad49183ec3b70de483912ee24ee706b
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/1ccd7d1b4ad49183ec3b70de483912ee24ee706b)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/execution/node/172/log)

### History
* :broken_heart: [Build #150573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/) failed 20c9bf9b9b06aab4d0f95024db35582ec683f276
* :broken_heart: [Build #150558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/) failed 8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/","number":"150558","commit":"8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/","number":"150573","commit":"20c9bf9b9b06aab4d0f95024db35582ec683f276"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/","number":"150634","commit":"1ccd7d1b4ad49183ec3b70de483912ee24ee706b"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/)
* Commit: 67a507ebc0395af5867e16ebbe88da528c4d5512
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Quick Commit Checks](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/execution/node/419/log)

### History
* :broken_heart: [Build #150634](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/) failed 1ccd7d1b4ad49183ec3b70de483912ee24ee706b
* :broken_heart: [Build #150573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/) failed 20c9bf9b9b06aab4d0f95024db35582ec683f276
* :broken_heart: [Build #150558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/) failed 8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/","number":"150558","commit":"8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/","number":"150573","commit":"20c9bf9b9b06aab4d0f95024db35582ec683f276"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/","number":"150634","commit":"1ccd7d1b4ad49183ec3b70de483912ee24ee706b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/","number":"150923","commit":"67a507ebc0395af5867e16ebbe88da528c4d5512"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/)
* Commit: 3b5a0e46795f0cc6caf37846d67faad31fcbb671
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/3b5a0e46795f0cc6caf37846d67faad31fcbb671)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/execution/node/719/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/execution/node/861/log)

### History
* :broken_heart: [Build #150923](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/) failed 67a507ebc0395af5867e16ebbe88da528c4d5512
* :broken_heart: [Build #150634](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/) failed 1ccd7d1b4ad49183ec3b70de483912ee24ee706b
* :broken_heart: [Build #150573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/) failed 20c9bf9b9b06aab4d0f95024db35582ec683f276
* :broken_heart: [Build #150558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/) failed 8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/","number":"150558","commit":"8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/","number":"150573","commit":"20c9bf9b9b06aab4d0f95024db35582ec683f276"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/","number":"150634","commit":"1ccd7d1b4ad49183ec3b70de483912ee24ee706b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/","number":"150923","commit":"67a507ebc0395af5867e16ebbe88da528c4d5512"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/","number":"150928","commit":"3b5a0e46795f0cc6caf37846d67faad31fcbb671"}]}
PIPELINE-->

VladLasitsa commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/)
* Commit: 7ec5db00f7d25cf87da59299d9fbe6b8b69235c2
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/7ec5db00f7d25cf87da59299d9fbe6b8b69235c2)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/eef094bafb4ce265126495754f8cb1cf760bb614) | [after](https://github.com/elastic/kibana/commit/7ec5db00f7d25cf87da59299d9fbe6b8b69235c2) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 540 | 442 | -98 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/eef094bafb4ce265126495754f8cb1cf760bb614) | [after](https://github.com/elastic/kibana/commit/7ec5db00f7d25cf87da59299d9fbe6b8b69235c2) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 1004.1KB | 789.3KB | -214.8KB |

### History
* :broken_heart: [Build #150928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/) failed 3b5a0e46795f0cc6caf37846d67faad31fcbb671
* :broken_heart: [Build #150923](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/) failed 67a507ebc0395af5867e16ebbe88da528c4d5512
* :broken_heart: [Build #150634](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/) failed 1ccd7d1b4ad49183ec3b70de483912ee24ee706b
* :broken_heart: [Build #150573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/) failed 20c9bf9b9b06aab4d0f95024db35582ec683f276
* :broken_heart: [Build #150558](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/) failed 8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150558/","number":"150558","commit":"8a4190c5f656e7e3e2a07d90e1b7845e6fe1510e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/","number":"150573","commit":"20c9bf9b9b06aab4d0f95024db35582ec683f276"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/","number":"150634","commit":"1ccd7d1b4ad49183ec3b70de483912ee24ee706b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/","number":"150923","commit":"67a507ebc0395af5867e16ebbe88da528c4d5512"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/","number":"150928","commit":"3b5a0e46795f0cc6caf37846d67faad31fcbb671"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/","number":"151203","commit":"7ec5db00f7d25cf87da59299d9fbe6b8b69235c2"}]}
PIPELINE-->

flash1293 commented on src/plugins/visualizations/server/embeddable/visualize_embeddable_factory.ts:
```diff
@@ -86,6 +94,9 @@ export const visualizeEmbeddableFactory = (): EmbeddableRegistryDefinition => {
           byValueMigrateTagcloud,
           byValueAddDropLastBucketIntoTSVBModel
         )(state),
+      '7.15.0': (state: SerializableRecord) => state,
```
This and the next line doesn't seem to do anything, are those leftovers?

VladLasitsa replied:
Removed lines with unneeded versions.

flash1293 commented on src/plugins/visualizations/server/migrations/visualization_saved_object_migrations.ts:
```diff
@@ -1121,4 +1145,7 @@ export const visualizationSavedObjectTypeMigrations = {
     replaceIndexPatternReference,
     addDropLastBucketIntoTSVBModel
   ),
+  '7.15.0': (doc: SavedObjectUnsanitizedDoc) => doc,
```
Same here

VladLasitsa replied:
As I know we cannot miss version here. We cannot just add `8.0.0`. But our logic should start work from `8.0.0`

flash1293 replied:
@alexwizp @VladLasitsa I tested this locally and couldn't get it to break even if these lines are omitted. Could you check how to replicate?

VladLasitsa replied:
Now I also cannot reproduce it, removed  line with unneeded versions.

flash1293 commented on x-pack/plugins/translations/translations/ja-JP.json:
```diff
@@ -4355,7 +4355,6 @@
     "visTypeTimeseries.markdown.editor.toggleEditorAriaLabel": "数列エディターを切り替える",
     "visTypeTimeseries.markdown.editor.variableNamePlaceholder": "変数名",
     "visTypeTimeseries.markdown.optionsTab.backgroundColorLabel": "背景色：",
-    "visTypeTimeseries.markdown.optionsTab.customCSSLabel": "カスタム CSS (Less をサポート) ",
```
AFAIK it's not necessary anymore to remove outdated translations - @stratoula can you confirm?

stratoula replied:
Yes my latest update was that we don't need to do it anymore.

VladLasitsa replied:
Okay, removed these changes

flash1293 commented on src/plugins/vis_type_timeseries/public/application/components/vis_types/markdown/vis.js:
```diff
@@ -65,16 +54,18 @@ function MarkdownVisualization(props) {
-              />
-            )}
-          </div>
+          {Style.it(
```
`style-it` is on old, unmaintained library which has security problems: https://github.com/elastic/kibana/issues/100639

It should be possible to do the same thing using `@emotion/react` - for example like this: https://github.com/elastic/kibana/blob/eef094bafb4ce265126495754f8cb1cf760bb614/src/plugins/chart_expressions/expression_tagcloud/public/expression_renderers/tagcloud_renderer.tsx#L60

VladLasitsa replied:
Hm, I'll try to use this

VladLasitsa replied:
Now I used `@emotion/react`

flash1293 reviewed (commented):
Added some comments, but this looks pretty well already

alexwizp commented on src/plugins/vis_type_timeseries/public/application/components/vis_types/markdown/vis.js:
```diff
@@ -42,13 +38,6 @@ function MarkdownVisualization(props) {
-      );
-    }
-
     const markdownClasses = classNames('kbnMarkdown__body', {
```
while replacing code with `@emotion`, try removing those awful classes. Maybe we can do it a little more professional

![image](https://user-images.githubusercontent.com/20072247/134036704-0e7dcde9-8b8b-424f-aaf8-0776ef51854a.png)

VladLasitsa replied:
Could you please provide ideas what we should do?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/)
* Commit: 44f65da61d069a1ac4c2732b54670e6ca067ed3a
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/execution/node/428/log)

### History
* :green_heart: [Build #151203](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/) succeeded 7ec5db00f7d25cf87da59299d9fbe6b8b69235c2
* :broken_heart: [Build #150928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/) failed 3b5a0e46795f0cc6caf37846d67faad31fcbb671
* :broken_heart: [Build #150923](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/) failed 67a507ebc0395af5867e16ebbe88da528c4d5512
* :broken_heart: [Build #150634](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/) failed 1ccd7d1b4ad49183ec3b70de483912ee24ee706b
* :broken_heart: [Build #150573](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/) failed 20c9bf9b9b06aab4d0f95024db35582ec683f276

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150573/","number":"150573","commit":"20c9bf9b9b06aab4d0f95024db35582ec683f276"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/","number":"150634","commit":"1ccd7d1b4ad49183ec3b70de483912ee24ee706b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/","number":"150923","commit":"67a507ebc0395af5867e16ebbe88da528c4d5512"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/","number":"150928","commit":"3b5a0e46795f0cc6caf37846d67faad31fcbb671"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/","number":"151203","commit":"7ec5db00f7d25cf87da59299d9fbe6b8b69235c2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/","number":"155812","commit":"44f65da61d069a1ac4c2732b54670e6ca067ed3a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/)
* Commit: fac28f3654e4d83b4aefef483d55df21cad51e4c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/fac28f3654e4d83b4aefef483d55df21cad51e4c)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ae4e7ccc517c75f55933333160145d11241d76a3) | [after](https://github.com/elastic/kibana/commit/fac28f3654e4d83b4aefef483d55df21cad51e4c) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 401 | 302 | -99 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ae4e7ccc517c75f55933333160145d11241d76a3) | [after](https://github.com/elastic/kibana/commit/fac28f3654e4d83b4aefef483d55df21cad51e4c) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 637.3KB | 482.7KB | -154.6KB |

### History
* :broken_heart: [Build #155812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/) failed 44f65da61d069a1ac4c2732b54670e6ca067ed3a
* :green_heart: [Build #151203](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/) succeeded 7ec5db00f7d25cf87da59299d9fbe6b8b69235c2
* :broken_heart: [Build #150928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/) failed 3b5a0e46795f0cc6caf37846d67faad31fcbb671
* :broken_heart: [Build #150923](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/) failed 67a507ebc0395af5867e16ebbe88da528c4d5512
* :broken_heart: [Build #150634](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/) failed 1ccd7d1b4ad49183ec3b70de483912ee24ee706b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150634/","number":"150634","commit":"1ccd7d1b4ad49183ec3b70de483912ee24ee706b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/","number":"150923","commit":"67a507ebc0395af5867e16ebbe88da528c4d5512"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/","number":"150928","commit":"3b5a0e46795f0cc6caf37846d67faad31fcbb671"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/","number":"151203","commit":"7ec5db00f7d25cf87da59299d9fbe6b8b69235c2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/","number":"155812","commit":"44f65da61d069a1ac4c2732b54670e6ca067ed3a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/","number":"155827","commit":"fac28f3654e4d83b4aefef483d55df21cad51e4c"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-vis-editors (Team:VisEditors)

stratoula commented:
@elasticmachine merge upstream

stratoula commented:
@VladLasitsa I think we can remove this `"less": "npm:@elastic/less@2.7.3-kibana"` from our package.json. Can you check and remove?

Also let's also close this https://github.com/elastic/kibana/issues/78422 with this PR as we are removing less
`

stratoula commented on src/plugins/vis_types/timeseries/public/application/components/panel_config/markdown.tsx:
```diff
@@ -275,8 +254,8 @@ export class MarkdownPanelConfig extends Component<
-                  defaultMessage="Custom CSS (supports Less)"
-                  description="CSS and Less are names of technologies and should not be translated."
+                  defaultMessage="Custom CSS"
+                  description="CSS is names of technologies and should not be translated."
```
nit: CSS is name of technology and...

VladLasitsa replied:
done

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/)
* Commit: 5da56e850dfa15d5fe9cf7509fb7157d547b0590
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/5da56e850dfa15d5fe9cf7509fb7157d547b0590)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/24bbf2cdfc3b4e1125d207bca4f96db073134313) | [after](https://github.com/elastic/kibana/commit/5da56e850dfa15d5fe9cf7509fb7157d547b0590) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 401 | 302 | -99 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/24bbf2cdfc3b4e1125d207bca4f96db073134313) | [after](https://github.com/elastic/kibana/commit/5da56e850dfa15d5fe9cf7509fb7157d547b0590) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 640.1KB | 485.4KB | -154.6KB |

### History
* :green_heart: [Build #155827](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/) succeeded fac28f3654e4d83b4aefef483d55df21cad51e4c
* :broken_heart: [Build #155812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/) failed 44f65da61d069a1ac4c2732b54670e6ca067ed3a
* :green_heart: [Build #151203](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/) succeeded 7ec5db00f7d25cf87da59299d9fbe6b8b69235c2
* :broken_heart: [Build #150928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/) failed 3b5a0e46795f0cc6caf37846d67faad31fcbb671
* :broken_heart: [Build #150923](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/) failed 67a507ebc0395af5867e16ebbe88da528c4d5512

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150923/","number":"150923","commit":"67a507ebc0395af5867e16ebbe88da528c4d5512"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/","number":"150928","commit":"3b5a0e46795f0cc6caf37846d67faad31fcbb671"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/","number":"151203","commit":"7ec5db00f7d25cf87da59299d9fbe6b8b69235c2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/","number":"155812","commit":"44f65da61d069a1ac4c2732b54670e6ca067ed3a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/","number":"155827","commit":"fac28f3654e4d83b4aefef483d55df21cad51e4c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/","number":"156633","commit":"5da56e850dfa15d5fe9cf7509fb7157d547b0590"}]}
PIPELINE-->

VladLasitsa commented:
> @VladLasitsa I think we can remove this `"less": "npm:@elastic/less@2.7.3-kibana"` from our package.json. Can you check and remove?
> 
> Also let's also close this #78422 with this PR as we are removing less
> 
> Finally, let's also fix our documentation (we don't support LESS anymore :D ) ![image](https://user-images.githubusercontent.com/17003240/135401956-8f860838-c27a-4a70-99d3-0e91a68f8c7d.png)

I fixed documentation and checked using of "less" lib. I haven't found any usage of it and removed it from out package.json and yarn.lock.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/)
* Commit: 0fe42ac79bc98985901119b17f376b38385ecde9
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/execution/node/165/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/execution/node/172/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/execution/node/124/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/execution/node/175/log)

### History
* :green_heart: [Build #156633](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/) succeeded 5da56e850dfa15d5fe9cf7509fb7157d547b0590
* :green_heart: [Build #155827](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/) succeeded fac28f3654e4d83b4aefef483d55df21cad51e4c
* :broken_heart: [Build #155812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/) failed 44f65da61d069a1ac4c2732b54670e6ca067ed3a
* :green_heart: [Build #151203](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/) succeeded 7ec5db00f7d25cf87da59299d9fbe6b8b69235c2
* :broken_heart: [Build #150928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/) failed 3b5a0e46795f0cc6caf37846d67faad31fcbb671

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150928/","number":"150928","commit":"3b5a0e46795f0cc6caf37846d67faad31fcbb671"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/","number":"151203","commit":"7ec5db00f7d25cf87da59299d9fbe6b8b69235c2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/","number":"155812","commit":"44f65da61d069a1ac4c2732b54670e6ca067ed3a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/","number":"155827","commit":"fac28f3654e4d83b4aefef483d55df21cad51e4c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/","number":"156633","commit":"5da56e850dfa15d5fe9cf7509fb7157d547b0590"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/","number":"156656","commit":"0fe42ac79bc98985901119b17f376b38385ecde9"}]}
PIPELINE-->

stratoula commented on yarn.lock:
```diff
@@ -6853,14 +6853,6 @@ ajv-keywords@^3.5.2:
   resolved "https://registry.yarnpkg.com/ajv-keywords/-/ajv-keywords-3.5.2.tgz#31f29da5ab6e00d1c2d329acf7b5929614d5014d"
   integrity sha512-5p6WTN0DdTGVQk6VjcEju19IgaHudalcfabD7yhDGeA6bcQnmL+CpveLJq/3hvfwd1aof6L386Ougkx6RfyMIQ==
 
-ajv@^4.9.1:
```
Interesting, why do we have so many changes in the yarn.lock file? 🤔

VladLasitsa replied:
Honestly, I don’t know, I just ran `kbn run bootstrap`

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156667/)
* Commit: 5cd4877af0999c9a598046fa5948a71c9e190751
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/5cd4877af0999c9a598046fa5948a71c9e190751)
* [Documentation Changes](https://kibana_110985.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/24bbf2cdfc3b4e1125d207bca4f96db073134313) | [after](https://github.com/elastic/kibana/commit/5cd4877af0999c9a598046fa5948a71c9e190751) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 401 | 302 | -99 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/24bbf2cdfc3b4e1125d207bca4f96db073134313) | [after](https://github.com/elastic/kibana/commit/5cd4877af0999c9a598046fa5948a71c9e190751) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 640.1KB | 485.4KB | -154.6KB |

### History
* :broken_heart: [Build #156656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/) failed 0fe42ac79bc98985901119b17f376b38385ecde9
* :green_heart: [Build #156633](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/) succeeded 5da56e850dfa15d5fe9cf7509fb7157d547b0590
* :green_heart: [Build #155827](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/) succeeded fac28f3654e4d83b4aefef483d55df21cad51e4c
* :broken_heart: [Build #155812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/) failed 44f65da61d069a1ac4c2732b54670e6ca067ed3a
* :green_heart: [Build #151203](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/) succeeded 7ec5db00f7d25cf87da59299d9fbe6b8b69235c2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151203/","number":"151203","commit":"7ec5db00f7d25cf87da59299d9fbe6b8b69235c2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/","number":"155812","commit":"44f65da61d069a1ac4c2732b54670e6ca067ed3a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/","number":"155827","commit":"fac28f3654e4d83b4aefef483d55df21cad51e4c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/","number":"156633","commit":"5da56e850dfa15d5fe9cf7509fb7157d547b0590"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/","number":"156656","commit":"0fe42ac79bc98985901119b17f376b38385ecde9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156667/","number":"156667","commit":"5cd4877af0999c9a598046fa5948a71c9e190751"}]}
PIPELINE-->

stratoula commented:
@elasticmachine merge upstream

stratoula reviewed (approved):
LGTM!! I have the exact same result with the `yarn kbn bootstrap`.
I tested it locally. It works fine 👏

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157038/)
* Commit: 251b974721f8ab99c42b8daaa0c6feb86e6bba53
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/251b974721f8ab99c42b8daaa0c6feb86e6bba53)
* [Documentation Changes](https://kibana_110985.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ff96b66eae26679f2ab79b02e104cfac098e25eb) | [after](https://github.com/elastic/kibana/commit/251b974721f8ab99c42b8daaa0c6feb86e6bba53) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 401 | 302 | -99 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ff96b66eae26679f2ab79b02e104cfac098e25eb) | [after](https://github.com/elastic/kibana/commit/251b974721f8ab99c42b8daaa0c6feb86e6bba53) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 640.1KB | 485.4KB | -154.6KB |

### History
* :green_heart: [Build #156667](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156667/) succeeded 5cd4877af0999c9a598046fa5948a71c9e190751
* :broken_heart: [Build #156656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/) failed 0fe42ac79bc98985901119b17f376b38385ecde9
* :green_heart: [Build #156633](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/) succeeded 5da56e850dfa15d5fe9cf7509fb7157d547b0590
* :green_heart: [Build #155827](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/) succeeded fac28f3654e4d83b4aefef483d55df21cad51e4c
* :broken_heart: [Build #155812](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/) failed 44f65da61d069a1ac4c2732b54670e6ca067ed3a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155812/","number":"155812","commit":"44f65da61d069a1ac4c2732b54670e6ca067ed3a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/","number":"155827","commit":"fac28f3654e4d83b4aefef483d55df21cad51e4c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/","number":"156633","commit":"5da56e850dfa15d5fe9cf7509fb7157d547b0590"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/","number":"156656","commit":"0fe42ac79bc98985901119b17f376b38385ecde9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156667/","number":"156667","commit":"5cd4877af0999c9a598046fa5948a71c9e190751"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157038/","number":"157038","commit":"251b974721f8ab99c42b8daaa0c6feb86e6bba53"}]}
PIPELINE-->

thomasneirynck reviewed (approved):
`maps/*` code review lgtm

alexwizp reviewed (approved):
LGTM! Tested locally

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157389/)
* Commit: 7b765957db485868df80087201b323936bad85e1
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-110985/7b765957db485868df80087201b323936bad85e1)
* [Documentation Changes](https://kibana_110985.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/7162f16f7a4d55be767d9c007cfb370e8d7709aa) | [after](https://github.com/elastic/kibana/commit/7b765957db485868df80087201b323936bad85e1) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 402 | 303 | -99 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7162f16f7a4d55be767d9c007cfb370e8d7709aa) | [after](https://github.com/elastic/kibana/commit/7b765957db485868df80087201b323936bad85e1) | diff |
| --- | --- | --- | --- |
| `visTypeTimeseries` | 640.7KB | 486.0KB | -154.6KB |

### History
* :green_heart: [Build #157038](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157038/) succeeded 251b974721f8ab99c42b8daaa0c6feb86e6bba53
* :green_heart: [Build #156667](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156667/) succeeded 5cd4877af0999c9a598046fa5948a71c9e190751
* :broken_heart: [Build #156656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/) failed 0fe42ac79bc98985901119b17f376b38385ecde9
* :green_heart: [Build #156633](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/) succeeded 5da56e850dfa15d5fe9cf7509fb7157d547b0590
* :green_heart: [Build #155827](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/) succeeded fac28f3654e4d83b4aefef483d55df21cad51e4c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @VladLasitsa

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155827/","number":"155827","commit":"fac28f3654e4d83b4aefef483d55df21cad51e4c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156633/","number":"156633","commit":"5da56e850dfa15d5fe9cf7509fb7157d547b0590"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156656/","number":"156656","commit":"0fe42ac79bc98985901119b17f376b38385ecde9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156667/","number":"156667","commit":"5cd4877af0999c9a598046fa5948a71c9e190751"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157038/","number":"157038","commit":"251b974721f8ab99c42b8daaa0c6feb86e6bba53"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157389/","number":"157389","commit":"7b765957db485868df80087201b323936bad85e1"}]}
PIPELINE-->
